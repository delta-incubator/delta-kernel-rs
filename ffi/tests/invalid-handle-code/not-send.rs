use delta_kernel_ffi_macros::handle_descriptor;

pub struct NotSend(*mut u32);

#[handle_descriptor(target=NotSend, mutable=false, sized=true)]
pub struct SharedNotSend;

#[handle_descriptor(target=NotSend, mutable=true, sized=true)]
pub struct MutNotSend;

fn main() {}
