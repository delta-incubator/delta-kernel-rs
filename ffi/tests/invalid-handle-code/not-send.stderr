error[E0277]: `*mut u32` cannot be sent between threads safely
 --> tests/invalid-handle-code/not-send.rs:5:28
  |
5 | #[handle_descriptor(target=NotSend, mutable=false, sized=true)]
  |                            ^^^^^^^ `*mut u32` cannot be sent between threads safely
  |
  = help: within `NotSend`, the trait `Send` is not implemented for `*mut u32`, which is required by `NotSend: Send`
note: required because it appears within the type `NotSend`
 --> tests/invalid-handle-code/not-send.rs:3:12
  |
3 | pub struct NotSend(*mut u32);
  |            ^^^^^^^
note: required by a bound in `delta_kernel_ffi::handle::HandleDescriptor::Target`
 --> src/handle.rs
  |
  |     type Target: ?Sized + Send;
  |                           ^^^^ required by this bound in `HandleDescriptor::Target`

error[E0277]: `*mut u32` cannot be sent between threads safely
 --> tests/invalid-handle-code/not-send.rs:8:28
  |
8 | #[handle_descriptor(target=NotSend, mutable=true, sized=true)]
  |                            ^^^^^^^ `*mut u32` cannot be sent between threads safely
  |
  = help: within `NotSend`, the trait `Send` is not implemented for `*mut u32`, which is required by `NotSend: Send`
note: required because it appears within the type `NotSend`
 --> tests/invalid-handle-code/not-send.rs:3:12
  |
3 | pub struct NotSend(*mut u32);
  |            ^^^^^^^
note: required by a bound in `delta_kernel_ffi::handle::HandleDescriptor::Target`
 --> src/handle.rs
  |
  |     type Target: ?Sized + Send;
  |                           ^^^^ required by this bound in `HandleDescriptor::Target`
